Deepsource fixes

kappq opened the pull request:
This fixes:
- Single character variable names
- Missing default case for switch statement
- Uses const for variable that are never reassigned

kappq committed:
Add custom status codes (#42)

* Add custom status codes

* Rename errors.py to codes.py

* Fix imports after module rename

This renames .errors to .codes.

* Remove useless argument

Removes the status code from RoomNotFound.

* Separate into different files

* Fix import after file separation

Rename .codes to .errors.

* Import StatusCode from codes

Montarex23 committed:
More modifiers (#29)

New modifiers:
  Boolean swap - change True to False and False to True
  Condition destroyer - replace == with = in if conditions
  Type chaos - mix keywords like int, str, bool
  Bracket hell - add or remove some brackets

kappq committed:
Data models (#39)

* Add models for event types

* Remove room_code from DisconnectData

* Add username to DisconnectData

* Add error event type

* Rename ConnectionData to ConnectData

* Add validator to Event

* Add status_code to Event

* Remove old event data classes

* Rename cls to self

Configures flake8 to ignore N805.

* Remove useless print statement

* Convert received data to Event

* Move ConnectionManager methods

* Add ConnectionManager.connect method

This method handles the connection type directly in the ConnectionManager.

* Convert ConnectionManager.connect to a static method

* Make _update_code_cache public

This also sets the line length for isort to 119.

* Make EventType a subclass of str

This helps with JSON serialization.

* Convert data type to Event

This also changes the docstrings of the modified functions.

* Update code cache tests

Updates the tests because _update_code_cache was made public.

* Match field order with docstring

* Add data attribute to RoomNotFoundError

* Edit Event validator to accept EventData

* Add difficulty validator

* Separate EventRequest and EventResponse

EventRequest is the data sent from the client to the server, EventResponse is the data sent from the server to the client. It allows to easily handle the status codes.

* Fix difficulty type annotation

* Fix EventRequest and EventResponse docstrings

* Remove useless docstrings

* Fix RuntimeError

This avoids that the client sends a message to itself after he disconnects.

LukeHankey committed:
feat: Add dev tools (#49)

* feat: Add more dev tools for development

* ci: Add flake8 plugins to flake8 pre-commit

* fix: flake8 errors fixed

kappq committed:
Use const for never reassigned variables

kappq committed:
Add default case for switch statement

kappq committed:
Change single character variable name

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey commented:
https://github.com/Vthechamp22/kindly-kappa/pull/26#discussion_r931646371 Can you resolve these comments as well whilst you're at it

LukeHankey committed:
Adds error handling on receive and create (#47)

* fix: TypeError was raised if an incorrect data type (non-json) was sent

* fix: Adds error handling to client.receive() for incorrect input types

New error codes (4002, 4003, 4004) to be sent back to the client and then disconnected.
Frontend will have to check if any of these error codes are sent back during a live game by checking that the event type is not connect

* feat: Add new error: RoomAlreadyExistsError

* Adds error handling for new error and corrects types

* Adds JSONDecodeError if no data is sent at all

* Remove VALIDATION_ERROR code as not used

* removes unnecessary await/async from ConnectionManager.connect

* Remove whitespace from StatusCodes

* fix grammar

* kappa changes

* fix event being None if an error occurs

Only disconnect if event is None, raised by either a WebsocketDisconnect or a RuntimeError
Return a valid EventRequest object in any other error case with default values which won't modify any code

LukeHankey committed:
feat: Adds a converter for the modified code (#51)

* feat: modify return structure

* test: Fix tests for output

* Change modifications detection

* remove print statement

* feat: Adds  the code modifier converter

* remove the deepdiff requirement

* fix flake8 errors

* Remove main guard

* Remove useless 'value' variable

Co-authored-by: Kappa <72104527+kappq@users.noreply.github.com>

kappq committed:
Project structure (#46)

* Separate Client and ConnectionManager in different files

* Change RoomData and ActiveRooms data structures

* Import from server instead of current package

* Use list as condition

This uses the list as a condition instead of checking if the length of the list is 0.

LukeHankey committed:
feat: Broadcast buggy code (#35)

* Import modifiers to main

* feat: Adds the modification to the manager class and broadcasts buggy data

* docs: reword docs for modify param

* Transform code from string into list of strings for use with each modfiier

* Introduce a new EventType (SEND_BUGS)

The frontend should periodically send this event to broadcast bugs back to the client

* Fix black and isort compatability

* Clean up logic for broadcasting bugs

* Remove the last '\n' as too many get added. Copy _list_of_lines

There is a copy of _list_of_lines for file_contents otherwise that and modified_contents reference the same object

* test: Fix tests

kappq committed:
Resolve Luke comments

LukeHankey commented:
Can resolve the 4 `Unexpected var, use let or const instead` fairly easily. Change them all to const as they aren't reassigned in their scope

kappq committed:
Remove unused variables

kappq committed:
Fix invalid variable declaration

kappq committed:
Change var for const

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
feat: Add Event handler (#55)

* Adds the username to the Client class

* Consistently sends back an EventRequest on receive

We now handle the WebsocketDisconnect and RuntimeError in the EventHandler class so we no longer need to return None

* Update typing for data param in broadcast to accept any subclass of EventData

* Rename data instance attribute to response to reduce ambiguity

* Add the EventHandler class

* refactor main to incorporate the EventHandler

* self.connection > self.manager

LukeHankey committed:
feat: Sync on join (#56)

* Adds the username to the Client class

* Consistently sends back an EventRequest on receive

We now handle the WebsocketDisconnect and RuntimeError in the EventHandler class so we no longer need to return None

* Update typing for data param in broadcast to accept any subclass of EventData

* Rename data instance attribute to response to reduce ambiguity

* Add the EventHandler class

* refactor main to incorporate the EventHandler

* Removes the connect method as EventHandler takes care of this now

* Update docs to be more accurate

* Update type for collaborators in SyncData

* Adds event handling logic for the sync event

* Fixed return type bug

Client.send expects an EventResponse (to include the event type) but the ConnectionManager.broadcast takes in and sent the EventData

* fix type issues with both type checkers

* Removed SEND_BUGS event till it's reworked

* forgot last assert statement

kappq committed:
Ignore node modules (#57)

LukeHankey committed:
Merge branch 'feature/vue-frontend' into deepsource-fixes

LukeHankey committed:
Ignore some deepsource warnings

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
spell skipcq correctly

LukeHankey committed:
Add documentation

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Last function documented. No more deepsource warnings!

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
No self.connection. Renamed to manager

LukeHankey committed:
Merge branch 'server-fixes' into feature/vue-frontend

LukeHankey committed:
Send back sync on initial connects to connect to room

LukeHankey committed:
Merge branch 'server-fixes' into feature/vue-frontend

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
Merge branch 'feature/vue-frontend' into deepsource-fixes

LukeHankey committed:
deepsource fixes

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
Merge branch 'feature/vue-frontend' into deepsource-fixes

LukeHankey committed:
Clear up invalid ref

LukeHankey committed:
Merge branch 'feature/vue-frontend' into deepsource-fixes
